src/api/api.rs

```diff
@@ -0,0 +1,70 @@
+use std::sync::{Arc, RwLock};
+
+use lazy_static::lazy_static;
+
+use crate::{
+    provider::{FeatureProvider, NoOpProvider, ProviderMetadata},
+    Client, EvaluationContext,
+};
+
+lazy_static! {
+    /// The singleton instance of [`OpenFeature`] struct.
+    /// The client should always use this instance to access OpenFeature APIs.
+    pub static ref SINGLETON: RwLock<OpenFeature> = RwLock::new(OpenFeature {
+        provider: Arc::new(NoOpProvider::default()),
+        evaluation_context: EvaluationContext::default()
+    });
+}
+
+/// THE struct of the OpenFeature API.
+/// Access it via the [`SINGLETON`] instance.
+pub struct OpenFeature {
+    provider: Arc<dyn FeatureProvider>,
+    evaluation_context: EvaluationContext,
+}
+
+impl OpenFeature {
+    pub fn new<T: FeatureProvider>(provider: T, evaluation_context: EvaluationContext) -> Self {
```

Comment:
Do we expect callers to call this, given there is a singleton that they should be using instead? If it's only for testing, we should wrap it in `#[cfg(test)]`

Reply:
If we can keep it private so only tests can manually create it that would probably be best. No reason to work outside the API. But if there is a use case please correct me here 🙂

Reply:
> If we can keep it private so only tests can manually create it that would probably be best. No reason to work outside the API. But if there is a use case please correct me here 🙂

Good idea. Fixed.